community/cypher/cypher-compiler-3.0/src/main/scala/org/neo4j/cypher/internal/compiler/v3_0/spi/QueryContext.scala

```diff
@@ -208,7 +198,6 @@ trait TransactionalContext[Graph,KernelStatement,StateView] {
 
   def commitAndRestartTx()
 
-  def graph: Graph
-
-  def stateView: StateView
+  // this should not be here, but it is needed by the code generation
```

Comment:
Can't you use the same trick we used before and have
```
type ReadOps
...
def readOperations: ReadOps
```
in `QueryContext` and then in the implementing class (`Neo4jTransactionContext`) you do

```
type ReadOps = org.neo4j.kernel.api.ReadOperations
...
def readOperations = statement.readOperations()
```?